httpbin/httpbin.go

```diff
@@ -179,3 +185,38 @@ func (h *HTTPBin) Handler() http.Handler {
 
 	return handler
 }
+
+func (h *HTTPBin) HeadersProcessor(headers http.Header) http.Header {
+	result := make(http.Header)
+	for k, v := range headers {
+		lowerCase := strings.ToLower(k)
+		skipped := false
+		for _, pattern := range h.excludeHeadersPatterns {
+			matched := pattern.Match([]byte(lowerCase))
+			if matched {
+				skipped = true
+				break
+			}
+		}
```

Comment:
I have a few ideas for optimizations here:
- Make this function return early if `h.excludeHeadersPatterns == nil`, which will be true in the vast majority of instances
- Build a single pattern that matches _all_ of excluded headers, so we only need to match each header key against a single pattern rather than this nested for loop
- Make sure the pattern matches case-insensitively so we do not need the separate `strings.ToLower()`